Modules/Devices/src/Devices.Application/Extensions/TierQueryableExtensions.cs

```diff
@@ -7,7 +7,7 @@ public static class TierQueryableExtensions
 {
     public static async Task<Tier> GetBasicTier(this IQueryable<Tier> query, CancellationToken cancellationToken)
     {
-        var basicTier = await query.FirstAsync(t => t.Name == TierName.BASIC_DEFAULT_NAME, cancellationToken);
-        return basicTier;
+        var basicTier = await query.FirstOrDefaultAsync(t => t.Name == TierName.BASIC_DEFAULT_NAME, cancellationToken);
+        return basicTier!;
```

Comment:
What makes sense to me here is either this (it should throw exception if not found)...
```chsarp
    public static async Task<Tier> GetBasicTier(this IQueryable<Tier> query, CancellationToken cancellationToken)
    {
        var basicTier = await query.FirstOrDefaultAsync(t => t.Name == TierName.BASIC_DEFAULT_NAME, cancellationToken);
        return basicTier ?? throw new Exception("Basic tier was not found.");
    }
```
...or this (basic tier is seeded and should therefore always exist):
```csharp
    public static async Task<Tier> GetBasicTier(this IQueryable<Tier> query, CancellationToken cancellationToken)
    {
        var basicTier = await query.FirstAsync(t => t.Name == TierName.BASIC_DEFAULT_NAME, cancellationToken);
        return basicTier;
    }
```
As none of the two work I reverted to the original state (with null warning suppression).